src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+import { Rundown } from '../models/rundown'
+import { BehaviorSubject, Observable, Subscription, SubscriptionLike } from 'rxjs'
+import { RundownStateService } from './rundown-state.service'
+import { RundownTimingService } from './rundown-timing.service'
+import { RundownTimingContext } from '../models/rundown-timing-context'
+import { Injectable } from '@angular/core'
+import { Segment } from '../models/segment'
+
+const HIGH_RESOLUTION_INTERVAL_DURATION_IN_MS: number = Math.floor(1000 / 30)
+const LOW_RESOLUTION_INTERVAL_DURATION_IN_MS: number = Math.floor(1000 / 4)
+
+@Injectable()
+export class RundownTimingContextStateService {
```

Comment:
FYI: The method order in this class is not a "linear read". 
The first usage of a method we encounter is `startLowResolutionTimer` from the constructor, but it's delcared as the fourth method. Similarly, `startHighResolutionTimer` is a private method that is declared before its first usage.

Reply:
Updated to have a linear flow.